bundles/org.openhab.binding.vigicrues/src/main/java/org/openhab/binding/vigicrues/internal/api/ApiHandler.java

```diff
@@ -0,0 +1,135 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.vigicrues.internal.api;
+
+import java.io.IOException;
+import java.time.ZonedDateTime;
+import java.util.Locale;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.vigicrues.internal.dto.hubeau.HubEauResponse;
+import org.openhab.binding.vigicrues.internal.dto.opendatasoft.OpenDatasoftResponse;
+import org.openhab.binding.vigicrues.internal.dto.vigicrues.CdStationHydro;
+import org.openhab.binding.vigicrues.internal.dto.vigicrues.InfoVigiCru;
+import org.openhab.binding.vigicrues.internal.dto.vigicrues.TerEntVigiCru;
+import org.openhab.binding.vigicrues.internal.dto.vigicrues.TronEntVigiCru;
+import org.openhab.core.i18n.TimeZoneProvider;
+import org.openhab.core.io.net.http.HttpUtil;
+import org.openhab.core.library.types.PointType;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * The {@link ApiHandler} is the responsible to call a given
+ * url and transform the answer in the appropriate dto class
+ *
+ * @author Gaël L'hopital - Initial contribution
+ */
+@NonNullByDefault
+public class ApiHandler {
+    private static final int TIMEOUT_MS = 30000;
+    private final Gson gson;
+
+    public ApiHandler(TimeZoneProvider timeZoneProvider) {
+        this.gson = new GsonBuilder().registerTypeAdapter(ZonedDateTime.class,
+                (JsonDeserializer<ZonedDateTime>) (json, type, jsonDeserializationContext) -> ZonedDateTime
+                        .parse(json.getAsJsonPrimitive().getAsString())
+                        .withZoneSameInstant(timeZoneProvider.getTimeZone()))
+                .create();
+    }
+
+    private String execute(String url) throws VigiCruesException {
+        String jsonResponse = "";
+        try {
+            jsonResponse = HttpUtil.executeUrl("GET", url, TIMEOUT_MS);
+            return jsonResponse;
+        } catch (IOException e) {
+            throw new VigiCruesException(e);
+        }
+    }
```

Comment:
You can simplify your api calls a bit by handling the json parsing in this method as well.

```suggestion
    private <T> T execute(String url, Class<T> responseType) throws VigiCruesException {
       try {
            String jsonResponse = HttpUtil.executeUrl("GET", url, TIMEOUT_MS);
            return gson.fromJson(response, responseType);
        } catch (IOException e) {
            throw new VigiCruesException(e);
        } catch (JsonSyntaxException e) {
            throw new VigiCruesException(e);
        }
    }
```